phi/tools/firecrawl.py

```diff
@@ -13,29 +16,39 @@ class FirecrawlTools(Toolkit):
     def __init__(
         self,
         api_key: Optional[str] = None,
+        api_url: Optional[str] = None,
         formats: Optional[List[str]] = None,
-        limit: int = 10,
+        crawl_limit: int = 10,
         scrape: bool = True,
         crawl: bool = False,
+        async_crawl: bool = False,
+        map: bool = False,
+        map_limit: int = 10,
     ):
         super().__init__(name="firecrawl_tools")
 
-        self.api_key: Optional[str] = api_key
+        self.api_key: Optional[str] = api_key or os.getenv("FIRECRAWL_API_KEY")
+        self.api_url: Optional[str] = api_url or os.getenv("FIRECRAWL_API_URL", "https://api.firecrawl.dev")
```

Comment:
Add an if condition to check if API key is available. If not then raise an error

Reply:
I noticed in other tools implementation, rather than `raise`, `logger.error` was used and within the methods  of the ToolKit also I noticed `return` rather than `raise` if `api_key` is not available.  this makes me believe the parent components using tools like `Assistant` class does not handle exception classes ?
If this is the case, I will make some minor changes at my end in the script as well.
Pls correct me if I am wrong or if I am missing here something. @manthanguptaa